### PR TITLE
Cell methods improvements

### PR DESCRIPTION
**Pull request by benjwadams**
Improve checking of cell methods contents which appear inside parentheses.  Currently these include standalone comments (no colon), interval specifications, and comments coming after standard elements.  Also checks if the element with a colon even exists and throws an error if it does not.

### CODE REVIEW

**Comment by benjwadams** [comment]
Hold on merging this, I need to fix for strings like "lat: lon: mean depth: variance (interval 0.02 degrees interval: 2 m comment: using spherical ellipsoid)"

File: compliance_checker/cf/cf.py [hunk 1]
```diff
@@ -2729,28 +2730,62 @@ def check_cell_methods(self, ds):
...
+                                # otherwise check that the comment is the last
+                                # item in the parentheses
+                                elif i != len(split_vals) - 1:
+                                    valid_modifier.messages.append('The non-standard "comment:" element must come after any standard elements in cell_methods for variable {}'.format(var.name))
+                                #
+                                else:
+                                    valid_modifier.score += 1
+                            else:
+                                valid_modifier.out_of += 1
+                                valid_modifier.messages.append('Invalid cell_methods keyword "{}" for variable {}. Must be one of [interval, comment]'.format(keyword, var.name))
```
Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: This is some awfully nested code, strongly recommend breaking the nested branches into functions.

Comment (benjwadams) [reply compliance_checker/cf/cf.py]: Will do.

File: compliance_checker/cf/cf.py [hunk 2]
```diff
@@ -2719,44 +2719,95 @@ def check_cell_methods(self, ds):
...
+        valid_info = TestCtx(BaseCheck.MEDIUM,
+                             '§7.3.3 {} has valid cell_methods modifiers'.format(var.name))
+        # if there are no colons, this is a simple comment
+        # TODO: are empty comments considered valid?
+        if ':' not in paren_contents:
+            valid_info.out_of += 1
+            valid_info.score += 1
+            return valid_info
+        # otherwise, split into k/v pairs
+        kv_pair_pat = r'(\S+:)\s+(.*(?=\s+\w+:)|[^:]+$)\s*'
```
Comment (benjwadams) [inline compliance_checker/cf/cf.py]: Not having quotation marks or commas like many key value pair specifications made this rather tricky to parse.  Hopefully this is robust.

File: compliance_checker/cf/cf.py [hunk 3]
```diff
@@ -2726,72 +2726,88 @@ def check_cell_methods(self, ds):
...
+        """
+        Checks that the spacing and/or comment info contained inside the
+        parentheses in cell_methods is well-formed
+        """
+        #valid_info = TestCtx(BaseCheck.MEDIUM,
+        #                            '§7.3.2 {} has valid cell_methods spacing/other info'.format(var.name))
+        valid_info = TestCtx(BaseCheck.MEDIUM,
+                             '§7.3.3 {} has valid cell_methods modifiers'.format(var.name))
+        # if there are no colons, this is a simple comment
+        # TODO: are empty comments considered valid?
```
Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: Yes, empty comments are valid, annoying, but valid.

**Pull request merged**